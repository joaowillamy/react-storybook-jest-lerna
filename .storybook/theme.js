import { create } from "@storybook/theming/create";

export default create({
  base: "light",

  colorPrimary: "#FDCB13",
  colorSecondary: "#18ACC4",

  // UI
  appBg: "white",
  appContentBg: "#F2F3F4",
  appBorderColor: "#F2F3F4",
  appBorderRadius: 4,

  // Typography
  fontBase: '"Montserrat", sans-serif',
  fontCode: "monospace",

  brandTitle: "Quero front test"
  // brandUrl: 'http://resolvvi.com.br/',
  // brandImage: ''
});
