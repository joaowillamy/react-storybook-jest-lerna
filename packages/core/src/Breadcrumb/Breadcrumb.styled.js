import styled from "styled-components";

import { chooseScrem } from "../help/chooseScreem";

export const Breadcrumb = styled.div`
  margin: 10px 0;
`;

export const BreadcrumbText = styled.p`
  font-weight: 600;
  font-size: 14px;
  ${chooseScrem}
`;

export const BreadcrumbTextBlue = styled.span`
  color: ${({ theme }) => theme.colors.bluePrimary};

  svg {
    width: ${({ theme }) => theme.iconSizes.lg};
  }
`;
