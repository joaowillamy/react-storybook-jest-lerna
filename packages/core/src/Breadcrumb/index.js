import React from "react";

import { Icon } from "../Icons";

import * as S from "./Breadcrumb.styled";

const Breadcrumb = props => {
  return (
    <S.Breadcrumb {...props}>
      <S.BreadcrumbText justDesktop>
        <S.BreadcrumbTextBlue> Home</S.BreadcrumbTextBlue> /{" "}
        <S.BreadcrumbTextBlue> Minha conta</S.BreadcrumbTextBlue> / Bolsas
        favoritas
      </S.BreadcrumbText>
      <S.BreadcrumbText justMobile>
        <S.BreadcrumbTextBlue>
          <Icon.ArrowBack /> Minha conta
        </S.BreadcrumbTextBlue>
      </S.BreadcrumbText>
    </S.Breadcrumb>
  );
};

export default Breadcrumb;
