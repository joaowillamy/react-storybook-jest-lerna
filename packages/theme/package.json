{
  "name": "@will-ui-kit/theme",
  "version": "1.0.2",
  "main": "lib/index.js",
  "module": "src/index.js"
}
