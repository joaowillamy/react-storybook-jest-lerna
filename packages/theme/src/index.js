export { default as ThemeProvider } from "./ThemeProvider";
export { default as DEFAULT_THEME } from "./theme";
export { GlobalStyles } from "./GlobalStyles";
