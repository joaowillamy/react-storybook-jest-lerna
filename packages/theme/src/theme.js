const BASE = 3;

const borders = {
  sm: color => `1px solid ${color || "#00000029"}`,
  md: color => `3px solid ${color || "#00000029"}`
};

const colors = {
  bluePrimary: "#18ACC4",
  blueSecondary: "#007A8D",
  yellowPrimary: "#FDCB13",
  yellowSecondary: "#DE9E1F",
  green: "#0FA866",
  black: "#1F2D30",
  grey: "#FBFBFB",
  Overlay: "#1F2D30e0" // com 88% de opacidade.
};

const fonts = {
  Montserrat: "Montserrat"
};

const fontSizes = {
  xs: "12px",
  sm: "14px",
  md: "16px",
  lg: "18px",
  xl: "22px",
  xxl: "26px",
  xxxl: "36px"
};

const fontWeights = {
  thin: 100,
  extralight: 200,
  light: 300,
  regular: 400,
  medium: 500,
  semibold: 600,
  bold: 700,
  extrabold: 800,
  black: 900
};

const iconSizes = {
  sm: "12px",
  md: "16px",
  lg: "26px"
};

const shadows = {
  sm: color => `0 1px 3px ${color || "#00000029"}`,
  md: color => `0 3px 4px ${color || "#00000029"}`,
  lg: color => `0 3px 6px ${color || "#00000029"}`
};

const space = {
  xs: `${BASE}px`,
  sm: `${BASE * 2}px`,
  md: `${BASE * 3}px`,
  lg: `${BASE * 4}px`,
  xl: `${BASE * 8}px`,
  xxl: `${BASE * 10}px`,
  xxxl: `${BASE * 12}px`
};

const DEFAULT_THEME = {
  borders,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  iconSizes,
  shadows,
  space
};

export default DEFAULT_THEME;
