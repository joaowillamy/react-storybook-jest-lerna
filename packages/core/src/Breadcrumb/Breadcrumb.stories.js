import React from "react";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Breadcrumb from ".";

export default {
  component: Breadcrumb,
  title: "Core|Breadcrumb"
};

export const breadcrumb = () => (
  <ThemeProvider>
    <Breadcrumb>{"Breadcrumb"}</Breadcrumb>
  </ThemeProvider>
);
