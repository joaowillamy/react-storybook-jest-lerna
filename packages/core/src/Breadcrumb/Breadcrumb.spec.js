import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { ThemeProvider } from "@joaowillamy-test-quero/theme";

import Breadcrumb from ".";

describe("Breadcrumb", () => {
  test("primary", () => {
    const tree = renderer
      .create(
        <ThemeProvider>
          <Breadcrumb>{"Test"}</Breadcrumb>
        </ThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
