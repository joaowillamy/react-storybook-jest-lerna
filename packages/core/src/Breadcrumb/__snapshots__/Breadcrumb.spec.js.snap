// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb primary 1`] = `
.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  margin: 10px 0;
}

.c1 {
  font-weight: 600;
  font-size: 14px;
}

.c3 {
  font-weight: 600;
  font-size: 14px;
}

.c2 {
  color: #18ACC4;
}

.c2 svg {
  width: 26px;
}

@media (max-width:768px) {
  .c1 {
    display: none;
  }
}

@media (min-width:768px) {
  .c3 {
    display: none;
  }
}

<div
  className="c0"
>
  <p
    className="c1"
  >
    <span
      className="c2"
    >
       Home
    </span>
     /
     
    <span
      className="c2"
    >
       Minha conta
    </span>
     / Bolsas favoritas
  </p>
  <p
    className="c3"
  >
    <span
      className="c2"
    >
      <svg
        aria-hidden="true"
        className="c4"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-name="Layer 2"
        >
          <path
            d="M13.83 19a1 1 0 01-.78-.37l-4.83-6a1 1 0 010-1.27l5-6a1 1 0 011.54 1.28L10.29 12l4.32 5.36a1 1 0 01-.78 1.64z"
            data-name="arrow-ios-back"
          />
        </g>
      </svg>
       Minha conta
    </span>
  </p>
</div>
`;
